pkg/client-nfd/compat/node-validator/node-validator_test.go

```diff
@@ -34,228 +34,283 @@ func init() {
 	fs.SetConfig(fs.NewConfig())
 }
 
+func buildDefaultSpec(rules []v1alpha1.Rule) *compatv1alpha1.Spec {
+	return &compatv1alpha1.Spec{
+		Version: compatv1alpha1.Version,
+		Compatibilties: []compatv1alpha1.Compatibility{
+			{
+				Description: "Fake compatibility",
+				Rules:       rules,
+			},
+		},
+	}
+}
+
+func buildDefaultExpectedOutput(status []ProcessedRuleStatus) []*CompatibilityStatus {
+	return []*CompatibilityStatus{
+		{
+			Description: "Fake compatibility",
+			Rules:       status,
+		},
+	}
+}
+
+func assertOutput(ctx context.Context, spec *compatv1alpha1.Spec, expectedOutput []*CompatibilityStatus) {
+	validator := New(
+		WithArgs(&Args{}),
+		WithArtifactClient(newMock(ctx, spec)),
+		WithSources(map[string]source.FeatureSource{fake.Name: source.GetFeatureSource(fake.Name)}),
+	)
+	output, err := validator.Execute(ctx)
+
+	So(err, ShouldBeNil)
+	So(output, ShouldEqual, expectedOutput)
+}
+
 func TestNodeValidator(t *testing.T) {
 	ctx := context.Background()
 
 	Convey("With a single compatibility set that contains flags, attributes and instances", t, func() {
-		spec := &compatv1alpha1.Spec{
-			Version: compatv1alpha1.Version,
-			Compatibilties: []compatv1alpha1.Compatibility{
+
+		Convey("That contains flag which results in match", func() {
```

Comment:
nit: with the new convey here, should the one above be changed to something like

```golang
Convey("With a single compatibility set", t func() {
```

Would it make more sense?